doc/scripts/gen_mon_command_api.py

```diff
@@ -0,0 +1,224 @@
+import os
+import json
+import io
+import textwrap
+import sys
+from subprocess import check_output
+
+header = '''
+.. This file is automatically generated. do not modify
+'''
+
+command_template = '''
+
+{prefix}
+{prefix_section_chars}
+
+{doc_string}
+'''
+
+footer = """
+"""
+
+
+def indent(s):
+    return '\n'.join(' ' * 4 + l for l in s.splitlines())
+
+
+class Flags:
+    NOFORWARD = (1 << 0)
+    OBSOLETE = (1 << 1)
+    DEPRECATED = (1 << 2)
+    MGR = (1 << 3)
+    POLL = (1 << 4)
+    HIDDEN = (1 << 5)
+
+    VALS = {
+        NOFORWARD: 'no_forward',
+        OBSOLETE: 'obsolete',
+        DEPRECATED: 'deprecated',
+        MGR: 'mgr',
+        POLL: 'poll',
+        HIDDEN: 'hidden',
+    }
+
+    def __init__(self, fs):
+        self.fs = fs
+
+    def __contains__(self, other):
+        return other in str(self)
+
+    def __str__(self):
+        keys = Flags.VALS.keys()
+        es = {Flags.VALS[k] for k in keys if self.fs & k == k}
+        return ', '.join(sorted(es))
+
+
+class CmdParam(object):
+    t = {
+        'CephInt': 'int',
+        'CephString': 'str',
+        'CephChoices': 'str',
+        'CephPgid': 'str',
+        'CephOsdName': 'str',
+        'CephPoolname': 'str',
+        'CephObjectname': 'str',
+        'CephUUID': 'str',
+        'CephEntityAddr': 'str',
+        'CephIPAddr': 'str',
+        'CephName': 'str',
+        'CephBool': 'bool',
+        'CephFloat': 'float',
+        'CephFilepath': 'str',
+    }
```

Comment:
For https://github.com/ceph/ceph/pull/34840 I've been thinking how to get this. As all Ceph Types are subclasses of `CephArgtype` we could simply exploit this and add an `@abstractproperty type` to the `CephArgtype`:
```py
class CephInt(CephArgtype):
    py_type = int

...
CEPH_TYPE_TO_PY_TYPE = {CephType.__class__ : CephType.py_type for CephType in CephArgType.__subclassess__}
```

Reply:
ok, for now, I'd like not to depend on ceph_argparse.py in the docs. the location is a bit awkward  and it has no setup.py, which makes it impossible to add ceph_argparse in the requirements.txt.